src/components/input-fields/field-context/FieldContextState.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useReducer } from "react";
+import { REGISTER_ADDON } from "./types";
+import FieldContextReducer from "./FieldContextReducer";
+import { FieldProvider, IFieldAddon, IFieldContext } from "./FieldContextContext";
+
+interface IFieldContextProvider {}
+
+const FieldContextState: React.FC<IFieldContextProvider> = ({ ...props }) => {
+  const initState = { addons: {} };
+
+  const [state, dispatch] = useReducer(FieldContextReducer, initState);
+
+  const registerAddon = (addon) => {
+    dispatch({
+      type: REGISTER_ADDON,
+      payload: addon,
+    });
+  };
+
+  const getAddonsList = () => {
+    const addonsList: IFieldAddon[] = [];
+
+    for (const addon in state.addons as IFieldContext) {
+      addonsList.push(state.addons[addon]);
+    }
+
+    return addonsList;
+  };
+
+  return (
+    <FieldProvider.Provider value={{ addons: state.addons, registerAddon, addonsList: getAddonsList() }}>
+      {React.Children.map(props.children, (child) => {
```

Comment:
@MoritzGruber @nicolasshiken what is the intention for this? Looks like this is setup for something that should be declarative and not state at all.

Reply:
This was a complete oopsie on my part. I was misusing our own API and fixed something unnecessary. Just removed the cloning stuff